File: src/main/java/at/fhhagenberg/sqe/api/ServicedFloorService.java
package at.fhhagenberg.sqe.api;

import at.fhhagenberg.sqe.entity.ServicedFloor;

import java.rmi.RemoteException;
import java.util.List;

public interface ServicedFloorService {
    public List<ServicedFloor> getAll(int elevatorNumber) throws RemoteException;
    public ServicedFloor get(int elevatorNumber, int floorNumber) throws RemoteException;
    public void updateServicedFloor(ServicedFloor servicedFloor) throws RemoteException;
}

File: src/main/kotlin/at/fhhagenberg/sqe/api/ServicedFloorService.kt
package at.fhhagenberg.sqe.api

import at.fhhagenberg.sqe.entity.ServicedFloor
import java.rmi.RemoteException

interface ServicedFloorService {
    @Throws(RemoteException::class)
    fun getAll(elevatorNumber: Int): List<ServicedFloor>

    @Throws(RemoteException::class)
    fun get(elevatorNumber: Int, floorNumber: Int): ServicedFloor?

    @Throws(RemoteException::class)
    fun updateServicedFloor(servicedFloor: ServicedFloor, isServiced: Boolean)
}